model/project_parser_s3.go

```diff
@@ -40,6 +40,7 @@ func NewParserProjectS3Storage(ppConf evergreen.ParserProjectS3Config) (*ParserP
 		Credentials: creds,
 	})
 	if err != nil {
+		utility.PutHTTPClient(c)
```

Comment:
In the success case (i.e. it returns the result), the HTTP client is needed to make S3 requests. In the error case, we need to clean it up because otherwise it'll leak HTTP clients.

This isn't unusual because the standard expectation is that we clean up in an error case (as the caller has no way to clean up a nil result), which we've always been expected to do.

Reply:
Yeah, we could just pass in the HTTP client into this constructor to get the defer, but it seemed like it would be slightly wonky because then we have to ensure the HTTP client is defer closed only _after_ all usages of the S3 implementation. It's a pretty small function and the usage of the err/success case is pretty straightforward, so I'm inclined to say that it's okay to do.